hoot-core/src/main/cpp/hoot/core/conflate/network/NetworkMergerCreator.cpp

```diff
@@ -399,39 +412,4 @@ bool NetworkMergerCreator::_isConflictingSet(const MatchSet& matches) const
   return conflicting;
 }
 
-// I dislike the nested for loop here - but whatcha gonna do? Maybe not create
-// duplicate matches in the first place
-void NetworkMergerCreator::_removeDuplicates(MatchSet& matches) const
-{
-  LOG_TRACE("Removing duplicate matches...");
-
-  for (MatchSet::iterator it = matches.begin(); it != matches.end(); ++it)
-  {
-    const NetworkMatch* nmi = dynamic_cast<const NetworkMatch*>(*it);
-    MatchSet::iterator jt = it;
-
-    for (++jt; jt != matches.end(); ++jt)
-    {
-      const NetworkMatch* nmj = dynamic_cast<const NetworkMatch*>(*jt);
-
-      if (nmi && nmj)
-      {
-        if (nmi->isVerySimilarTo(nmj))
-          LOG_TRACE(nmi->getEdgeMatch()->getUid() << " is very similar to " << nmj->getEdgeMatch()->getUid());
-
-        if (nmi->contains(nmj))
-          LOG_TRACE(nmi->getEdgeMatch()->getUid() << " contains " << nmj->getEdgeMatch()->getUid());
-
-        if (nmi->isVerySimilarTo(nmj))
-        {
-          MatchSet::iterator tmp = jt;
-          ++tmp;
-          matches.erase(jt);
-          jt = tmp;
-        } // end if very similar
-      } // end if valid pointers
-    } // inner for
-  } // outer for
-}
-
```

Comment:
This is great.  I think the code seems simpler when not adding extra matches and removing them later.  The speed-ups for network conflation that you've been able to achieve are very impressive.